tests/integration-service/namespace-backed-environments.go

```diff
@@ -197,14 +199,22 @@ var _ = framework.IntegrationServiceSuiteDescribe("Namespace-backed Environment
 		})
 
 		It("checks for GitOpsDeployments after Ephemeral env has been created", func() {
-			godl, err = f.AsKubeAdmin.GitOpsController.ListAllGitOpsDeployments(testNamespace)
-			Expect(err).ToNot(HaveOccurred())
-			Expect(godl.Items).ToNot(BeNil())
+			Eventually(func() error {
+				godl, err = f.AsKubeAdmin.GitOpsController.ListAllGitOpsDeployments(testNamespace)
+				if godl.Items == nil || len(godl.Items) == 0 {
+					return fmt.Errorf("No GitOpsDeployments found, GitOpsDeployments is empty")
+				}
+				if !reflect.ValueOf(godl.Items[0].Status).IsZero() && reflect.ValueOf(godl.Items[0].Status.ReconciledState).IsZero() {
+					return fmt.Errorf("ReconciledState doesn't exist yet for GitOpsDeployment.")
+				}
+				return err
+			}, time.Minute*1, time.Second*1).Should(Succeed(), fmt.Sprintf("timed out checking GitOpsDeployments after Ephemeral Environment was created %s", ephemeralEnvironment.Name))
+
 			Expect(godl.Items[0].Spec.Source.RepoURL).ToNot(BeEmpty())
 			Expect(godl.Items[0].Spec.Source.Path).ToNot(BeEmpty())
 			Expect(godl.Items[0].Spec.Source.TargetRevision).To(Equal("main"))
 			Expect(godl.Items[0].Spec.Type).To(Equal("automated"))
-			Expect(godl.Items[0].Status.ReconciledState).ToNot(BeNil())
+			Expect(err).ToNot(HaveOccurred())
```

Comment:
hmm I'm not sure under what circumstances this assertion would ever fail... does it sound reasonable to move it directly under `ListAllGitOpsDeployments` function call?

Reply:
Expect(err).ToNot(HaveOccurred()) moved